src/func/hashing.rs

```diff
@@ -30,31 +34,26 @@ pub const ALT_ZERO_HASH: u64 = 42;
 ///
 /// Panics when hashing any data type other than a [`u64`].
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
-struct StraightHasher(u64);
+pub struct StraightHasher(u64);
 
 impl Hasher for StraightHasher {
     #[inline(always)]
     fn finish(&self) -> u64 {
         self.0
     }
     #[inline]
-    fn write(&mut self, bytes: &[u8]) {
-        assert_eq!(bytes.len(), 8, "StraightHasher can only hash u64 values");
-
-        let mut key = [0_u8; 8];
-        key.copy_from_slice(bytes);
-
-        self.0 = u64::from_ne_bytes(key);
+    fn write(&mut self, _bytes: &[u8]) {
+        panic!("StraightHasher can only hash u64 values");
+    }
 
-        if self.0 == 0 {
```

Comment:
You'd need the ALT_ZERO_HASH because the engine assumes a zero to be "no hash".  Essentially zero is mapped to `None`.